### PR TITLE
docs - update full-app embedding

### PR DESCRIPTION
**Pull request by nllho**
- Re-worked the original doc so that the info is organized by use case instead.
- Removed "Appearance" from embedding/start (it felt out of place -- Appearance it doesn't show up under Embedding in the sidebar or under "On this page"). I made sure to link to Appearance throughout the embedding docs, though!

### CODE REVIEW

File: docs/embedding/full-app-embedding.md [hunk 1]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
+4. Under **Authorized origins**, add the URL of the website or web app where you want to embed Metabase.
 
-Once you do, you'll see a set of options:
+## Setting up embedding on your website
 
-- **Embedding secret key:** You can ignore this setting, which is only for standalone chart or dashboard embeds.
+1. Create an iframe with a `src` attribute set to:
+   - the [URL](#pointing-an-iframe-to-a-metabase-url) of the Metabase page you want to embed, or
+   - an [authentication endpoint](#pointing-an-iframe-to-an-authentication-endpoint) that redirects to your Metabase URL.
+2. Optional: Depending on the way your web app is set up, set [environment variables](../configuring-metabase/environment-variables.md) to:
```
Comment (nllho) [inline docs/embedding/full-app-embedding.md:36]: @flamber @paoliniluis @Tony-metabase 

Could you guys check the correctness of the sections under 2 (env vars) and 3 (JavaScript stuff) by the end of the week?

I did lots of reading, but still have the least confidence in translating those use cases correctly.

If there's no comments by the end of the week, I'll assume that it's good to go :) Thanks!

File: docs/embedding/full-app-embedding.md [hunk 2]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
-- Optional: JavaScript using [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) in your application for enabling communication to and from the embedded Metabase. Here are the types of `postMessage` messages we currently support:
+```
+http://metabase.yourcompany.com/dashboard/1
+```
 
-## Supported `postMessage` messages _from_ embedded Metabase:
+### Pointing an iframe to an authentication endpoint
 
-- `frame` message with `normal` mode: the current page in the embedded Metabase will fill whatever size `iframe` it is displayed in, e.x. the `/question` pages:
+To embed an auth endpoint with a redirect to your Metabase URL, you'll need to double encode the Metabase URL and add it as a parameter:
```
Comment (nllho) [inline docs/embedding/full-app-embedding.md:66]: @nllho to update the example here, I don't think it's actually double encoded.

File: docs/embedding/full-app-embedding.md [hunk 3]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
+
+And, to send a "location" message _to_ your embedded Metabase:
 
-The header, visible by default, only applies to dashboards and questions. The header refers to the info and buttons above a question or dashboard. The header includes the title, additional info, and the action buttons (Filter and Summarize).
+```
+{ “metabase”: { “type”: “location”, “location”: LOCATION_OBJECT_OR_URL }}
+```
 
-If you enable the header, these sub-components are visible by default:
+## Showing or hiding Metabase UI components
```
Comment (nllho) [inline docs/embedding/full-app-embedding.md:138]: @nllho to mention double encoding here as well, re: https://metabase.zendesk.com/agent/tickets/12156

File: docs/embedding/full-app-embedding.md [hunk 4]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
 
-- A [paid plan of Metabase](https://www.metabase.com/pricing) that includes full-app embedding.
-- A separate web application that you want to embed your dashboards and charts in.
+If you're dealing with a [multi-tenant](https://www.metabase.com/learn/customer-facing-analytics/multi-tenant-self-service-analytics) situation, check out our recommendations for [Configuring permissions for different customer schemas](https://www.metabase.com/learn/permissions/multi-tenant-permissions).
 
-## Enabling embedding in Metabase
+## Enabling full-app embedding in Metabase
 
-First, let's enable embedding in your Metabase instance. Go to the Admin Panel, and under Settings, go to the “Embedding in other applications” tab. From there, click “Enable.”
+1. Go to **Admin settings** > **Embedding**.
```
Comment (jeff-bruemmer) [inline docs/embedding/full-app-embedding.md:26]:
```suggestion
1. Go to **Admin settings** > **Settings** > **Embedding**.
```

Probably not necessary, but doesn't hurt.

File: docs/embedding/full-app-embedding.md [hunk 5]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
+2. Click **Enable**.
+3. Click **Full-app embedding**.
+4. Under **Authorized origins**, add the URL of the website or web app where you want to embed Metabase.
```
Comment (jeff-bruemmer) [inline docs/embedding/full-app-embedding.md:29]:
```suggestion
4. Under **Authorized origins**, add the URL of the website or web app where you want to embed Metabase (e.g., `https://*.example.com`).
```
We do mention the example in the field itself; the issue is that they might paste an incorrectly formatted string in there first, and may miss the example.

File: docs/embedding/full-app-embedding.md [hunk 6]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
+Metabase offers several [types of embedding](./introduction.md) with different levels of customization and security.
 
-You'll be putting the whole Metabase app into an iframe, and the SSO integration you've set up with Metabase will be used to make sure the embedded Metabase respects the collection and data permissions you've set up for your user groups. Clicking on charts and graphs in the embed will do just what they do in Metabase itself. You can even display a specific Metabase collection in an embed to allow your users to browse through all the dashboards and questions that you've made available to them. The only difference is that Metabase's top nav bar and global search will not be rendered in your iframe.
+**Full-app embedding** is the only type of embedding that integrates with your [permissions](../permissions/introduction.md) and [SSO](../people-and-groups/start.md#authentication) to give people the right level of access to [query](https://www.metabase.com/glossary/query_builder) and [drill-down](https://www.metabase.com/glossary/action_menu) into your data.
```
Comment (jeff-bruemmer) [inline docs/embedding/full-app-embedding.md:13]: Good clarification here. We may want to explicitly explain how full-app embedding differs from signed embedding.

File: docs/embedding/full-app-embedding.md [hunk 7]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
-  { "metabase": { "type": "frame", "frame": { "mode": "normal" }}}
+```
+http://yourcompany.com/api/auth?redirect=http%3A%2F%2Fmetabase.yourcompany.com%2Fdashboard%2F1
+```
 
-- `frame` message with `fit` mode: the current page in the embedded Metabase expects a specific aspect ratio, e.x. `/dashboard` pages, so the `iframe` should be set to specified height if you don't want it to scroll:
+## Embedding Metabase in a different domain
 
-  { "metabase": { "type": "frame", "frame": { "mode": "fit", height: HEIGHT_IN_PIXELS }}}
+If you want to embed Metabase in another domain (e.g., Metabase is hosted at `metabase.yourcompany.com`, but you want to embed Metabase at `yourcompany.github.io`), set the following environment variable:
```
Comment (jeff-bruemmer) [inline docs/embedding/full-app-embedding.md:74]:
```suggestion
If you want to embed Metabase in another domain (e.g., Metabase is hosted at `metabase.yourcompany.com`, but you want to embed Metabase at `yourcompany.github.io`), set the following [environment variable](../configuring-metabase/environment-variables.md):
```

File: docs/embedding/full-app-embedding.md [hunk 8]
```diff
@@ -4,161 +4,227 @@ redirect_from:
...
-E.g., the URL
+To automatically clear a person's login cookies when they end a browser session:
 
 ```
-http://localhost:3000/?top_nav=false&side_nav=false
+MB_SESSION_COOKIES=true
 ```
 
-would disable the top and side nav:
+To log everyone out of Metabase, load the following URL (for example, in a hidden iframe on the logout page of your application):
```
Comment (jeff-bruemmer) [inline docs/embedding/full-app-embedding.md:100]: Log "everyone" out? Or someone out?

File: docs/embedding/signed-embedding.md [hunk 9]
```diff
@@ -112,7 +110,7 @@ The banner appears on signed embeds created with the Metabase open source versio
 
 ## Regenerating the secret key
 
-Your embedding secret key is used to sign JWTs for all of your [embedding URLs](#adding-the-embedding-url-to-your-website). You can find it under **Admin settings** > **Embedding in other Applications**.
+Your embedding secret key is used to sign JWTs for all of your [embedding URLs](#adding-the-embedding-url-to-your-website). You can find it under **Admin settings** > **Embedding**.
```
Comment (jeff-bruemmer) [inline docs/embedding/signed-embedding.md:113]:
```suggestion
Your embedding secret key is used to sign JWTs for all of your [embedding URLs](#adding-the-embedding-url-to-your-website). You can find your embedding secret key by clicking on the **gear** icon in the upper right and navigating to **Admin settings** > **Settings** > **Embedding**.
```

**Review by jeff-bruemmer — APPROVED** [review]
Great revision, @nllho. Good clarifications (and nice job axing some images).

**Pull request merged**